### ISSUE TITLE
Wordwrap in buttons

### ISSUE THREAD

**Issue by neilsimp1** [ppizarror/pygame-menu#488]
**Is your feature request related to a problem? Please describe.**
I'm trying to handle cases in my program where the text for a button might be a bit long.
I'm able to `button.set_max_width()` but this only scales the text in order to make it fit.

**Describe the solution you'd like**
I'd like to be able to `wordwrap=True` similar to how a `Label` works. This would increase the height of the button and allow text to flow onto multiple lines.

**Describe alternatives you've considered**
Creating a custom widget that combines the functionality of Buttons and Labels.

**Comment by neilsimp1** [comment]
Please se #490 :)

**Comment by ppizarror** [comment]
Great contribution! Please add yourself to docs/_source/contributors.rst. Will look tomorrow and upload to pip.

–
Pablo

**Comment by neilsimp1** [comment]
Done. Thanks for considering my code :)

**Comment by ppizarror** [comment]
Hi. I uploaded v4.4.9 to pip.
![image](https://github.com/user-attachments/assets/95cd31fa-ed0e-4a08-b4de-38adf5f7400a)
I changed the architecture of your enhancement a bit. I noticed that most code relied on ``Label, `` so I inherited it from Button. I always prefer simpler approaches or detect if there are any refactor patterns in the code. This project has a lot of tests, so we are backed.
Also, I added a test for this feature.

**Comment by neilsimp1** [comment]
Thank you! I was worried about repeating code but wasn't confident enough to do such a large refactor.
___

Unfortunately, after pulling in the latest copy and trying my project, I run in to my original error (the width of the button is wider than that of the frame surrounding it).

It doesn't seem to be doing the word wrap and I think I see why:
https://github.com/ppizarror/pygame-menu/blob/master/pygame_menu/widgets/widget/label.py#L229C13-L229C22
Here, it's getting the max_container_width by looking at the widget's menu width.

In my case, these buttons are within a VFrame, and I need them to not exceed the width of that.

It's hard to see in my screenshot before so here's a screenshot outlining things. The green is a menu, blue are frames.
![image](https://github.com/user-attachments/assets/ace4c775-c697-4693-855a-5e685c723884)

I've updated my code to call the Button without `kwargs`, and you can see the relevant code here: https://gitlab.com/neilsimp1/pacab/-/blob/main/src/menus/inventorymenu.py?ref_type=heads#L58


Do you have any suggestion if anything can be done on my end to make this look for the frame width instead of the menu width?
Or, does anything need to change in `_get_max_container_width()` to make it aware of Frames?

**Comment by ppizarror** [comment]
As ``Button`` passes down all ``*args`` and ``*kwargs`` I dediced to remove accepting kwargs from button. ``ButtonManager`` processes all arguments and helps to pass them down to the widget.

Also, I fixed the Frame awareness while computing container width. See https://github.com/ppizarror/pygame-menu/commit/96f8a7ab953bb72a96fad246061092f54ed94231. Let me know if that works, so I upload new v4.5.1 to pip.
